domain/ide/workspace/folder.go

```diff
@@ -134,20 +135,53 @@ func (f *Folder) DocumentDiagnosticsFromCache(file string) []snyk.Issue {
 }
 
 func (f *Folder) processResults(issues []snyk.Issue) {
-	var issuesByFile = map[string][]snyk.Issue{}
+	issuesByFile := map[string][]snyk.Issue{}
 	dedupMap := f.createDedupMap()
 
 	// TODO: perform issue diffing (current <-> newly reported)
+	// Update diagnostic cache
+	var cachedIssues interface{}
 	for _, issue := range issues {
-		cachedIssues := f.documentDiagnosticCache.Get(issue.AffectedFilePath)
+		cachedIssues = f.documentDiagnosticCache.Get(issue.AffectedFilePath)
 		if cachedIssues == nil {
 			cachedIssues = []snyk.Issue{}
 		}
+
 		if !dedupMap[f.getUniqueIssueID(issue)] {
 			cachedIssues = append(cachedIssues.([]snyk.Issue), issue)
 		}
+
 		f.documentDiagnosticCache.Put(issue.AffectedFilePath, cachedIssues)
-		issuesByFile[issue.AffectedFilePath] = cachedIssues.([]snyk.Issue)
+	}
+
+	// update issues by file
+	filteredIssues := []snyk.Issue{}
+	severityFilters := config.CurrentConfig().FilterSeverity()
+	// prevent panic: interface conversion: interface {} is nil, not []snyk.Issue
+	if cachedIssues == nil {
```

Comment:
So, in which constellation can this be nil? If there is more than one issue to be processed, the array is created in line 147. If `len(issues)` is 0, why are we even processing? Shouldn't we just early end the processing and return

Reply:
It looks like this was a remnant from earlier in the development, just tried it without this check and it's working fine 👍 
If `len(issues) == 0` then we'd skip the filtering and process the diagnostics/hovers as we did previously.